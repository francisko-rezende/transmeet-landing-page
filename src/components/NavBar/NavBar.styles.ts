import styled from 'styled-components'

export const Nav = styled.nav`
  max-width: 40rem;
`

export const List = styled.ul`
  display: grid;
  grid-template-columns: repeat(4, minmax(0, 1fr));
  justify-content: center;
  align-items: center;
  list-style: none;
  /* gap: 35px; */
`

export const Item = styled.li`
  width: 10rem;
  text-align: center;
`

export const NavLink = styled.a`
  color: ${({ theme }) => theme.colors.typographyColors.white};
  text-decoration: none;
  position: relative;
  width: 100%;
  margin: 0 auto;
  font-size: ${({ theme }) => theme.font.sizes['4']};

  &:hover {
    font-weight: ${({ theme }) => theme.font.weight.bold};

    &::after {
      content: '';
      position: absolute;
      display: block;
      bottom: -1rem;
      height: 0.5rem;
      background: linear-gradient(
        269.71deg,
        rgba(255, 255, 255, 0) 0.01%,
        #f7c2d2 19.66%,
        #bb4981 49%,
        #185aa1 74.55%,
        #1f478b 100%
      );
      animation: hoverAnimation 0.3s forwards;
    }

    @keyframes hoverAnimation {
      from {
        width: 0;
        left: 50%;
      }
      to {
        width: 100%;
        left: 0;
      }
    }
  }
`
