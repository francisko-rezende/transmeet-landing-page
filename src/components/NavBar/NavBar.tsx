import React from 'react'

import Link from 'next/link'

import * as S from './NavBar.styles'

const NavBar = () => {
  return (
    <S.Nav>
      <S.List>
        <S.Item>
          <Link href="#" passHref>
            <S.NavLink>Home</S.NavLink>
          </Link>
        </S.Item>
        <S.Item>
          <Link href="/" passHref>
            <S.NavLink>Aplicativo</S.NavLink>
          </Link>
        </S.Item>
        <S.Item>
          <Link href="/about" passHref>
            <S.NavLink>Sobre</S.NavLink>
          </Link>
        </S.Item>
        <S.Item>
          <Link href="/contact" passHref>
            <S.NavLink>Contato</S.NavLink>
          </Link>
        </S.Item>
      </S.List>
    </S.Nav>
  )
}

export default NavBar
