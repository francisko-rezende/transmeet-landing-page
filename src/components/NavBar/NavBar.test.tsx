import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import NavBar from '.'

describe('<NavBar />', () => {
  test('should render the heading', () => {
    const { container } = renderWithTheme(<NavBar />)

    expect(screen.getByRole('link', { name: /Home/i })).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /Aplicativo/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /sobre/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /contato/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
