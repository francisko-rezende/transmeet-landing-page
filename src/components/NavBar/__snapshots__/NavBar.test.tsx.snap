// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render the heading 1`] = `
.c0 {
  max-width: 40rem;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(4,minmax(0,1fr));
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style: none;
}

.c2 {
  width: 10rem;
  text-align: center;
}

.c3 {
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  width: 100%;
  margin: 0 auto;
  font-size: 16px;
}

.c3:hover {
  font-weight: 700;
}

.c3:hover::after {
  content: '';
  position: absolute;
  display: block;
  bottom: -1rem;
  height: 0.5rem;
  background: linear-gradient( 269.71deg, rgba(255,255,255,0) 0.01%, #f7c2d2 19.66%, #bb4981 49%, #185aa1 74.55%, #1f478b 100% );
  -webkit-animation: hoverAnimation 0.3s forwards;
  animation: hoverAnimation 0.3s forwards;
}

<nav
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/application"
      >
        Aplicativo
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/about"
      >
        Sobre
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/contact"
      >
        Contato
      </a>
    </li>
  </ul>
</nav>
`
